src/test/java/no/unit/nva/handlers/UpdateUserHandlerTest.java

```diff
@@ -1,65 +1,186 @@
 
+import static no.unit.nva.database.DatabaseServiceImpl.USER_NOT_FOUND_MESSAGE;
+import static no.unit.nva.handlers.UpdateUserHandler.INCONSISTENT_USERNAME_IN_PATH_AND_OBJECT_ERROR;
+import static no.unit.nva.handlers.UpdateUserHandler.LOCATION_HEADER;
+import static no.unit.nva.handlers.UpdateUserHandler.NO_USERS_PATH_PARAMETER_FOUND_ERROR_MESSAGE;
+import static no.unit.nva.handlers.UpdateUserHandler.USERNAME_PATH_PARAMETER;
+import static nva.commons.utils.JsonUtils.objectMapper;
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.collection.IsMapContaining.hasKey;
 import static org.hamcrest.core.Is.is;
 import static org.hamcrest.core.IsEqual.equalTo;
+import static org.hamcrest.core.StringContains.containsString;
 import static org.mockito.Mockito.mock;
 
 import com.amazonaws.services.lambda.runtime.Context;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.InvocationTargetException;
 import java.util.Collections;
+import java.util.Map;
 import no.unit.nva.database.DatabaseAccessor;
 import no.unit.nva.database.DatabaseServiceImpl;
 import no.unit.nva.exceptions.ConflictException;
-import no.unit.nva.exceptions.InvalidInputException;
 import no.unit.nva.exceptions.InvalidEntryInternalException;
+import no.unit.nva.exceptions.InvalidInputException;
 import no.unit.nva.exceptions.NotFoundException;
 import no.unit.nva.model.RoleDto;
 import no.unit.nva.model.UserDto;
+import no.unit.nva.testutils.HandlerRequestBuilder;
+import no.unit.nva.utils.EntityUtils;
 import nva.commons.exceptions.ApiGatewayException;
-import nva.commons.handlers.RequestInfo;
+import nva.commons.handlers.GatewayResponse;
+import org.apache.http.HttpStatus;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
+import org.zalando.problem.Problem;
 
 public class UpdateUserHandlerTest extends DatabaseAccessor {
 
     public static final String SAMPLE_ROLE = "someRole";
     public static final String SAMPLE_USERNAME = "some@somewhere";
     public static final String SAMPLE_INSTITUTION = "somewhere";
     public static final String ANOTHER_ROLE = "ANOTHER_ROLE";
+    public static final String SOME_OTHER_USERNAME = "SomeOtherUsername";
     private DatabaseServiceImpl databaseService;
     private Context context;
 
+    private ByteArrayOutputStream output;
+
     @BeforeEach
     public void init() {
         databaseService = new DatabaseServiceImpl(initializeTestDatabase(), envWithTableName);
         context = mock(Context.class);
+        output = new ByteArrayOutputStream();
     }
 
-    @DisplayName("processInput() returns updated user when path contains an exisitng user id, "
-        + "body contains a valid UserDto and path id is the same as the body id ")
+    @DisplayName("handleRequest() returns Location header with the URI to updated user when path contains "
```

Comment:
Better to test handleRequest than process input because this is the functionality that we are interested in. processinput is more of an implementation detail.